packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,397 @@
+import { pipe, flow } from "fp-ts/function"
+import * as TE from "fp-ts/TaskEither"
+import * as E from "fp-ts/Either"
+import * as O from "fp-ts/Option"
+import * as A from "fp-ts/Array"
+import * as RA from "fp-ts/ReadonlyArray"
+import { OpenAPIV3 } from "openapi-types"
+import isPlainObject from "lodash/isPlainObject"
+import omit from "lodash/omit"
+import { HoppExporter } from "."
+import { tupleToRecord } from "~/helpers/functional/record"
+import { safeParseJSON } from "~/helpers/functional/json"
+import { objHasProperty } from "~/helpers/functional/object"
+
+const convertHoppFormDataEntryToOAFormDataEntry = (
+  entry: FormDataKeyValue
+): [string, OpenAPIV3.SchemaObject] =>
+  entry.isFile
+    ? [entry.key, { type: "string", format: "binary" }]
+    : [entry.key, { type: "string", default: entry.value }]
+
+const generateFormDataEntries = (
+  formdataBody: HoppRESTReqBodyFormData
+): OpenAPIV3.SchemaObject["properties"] =>
+  pipe(
+    formdataBody.body,
+    A.map(convertHoppFormDataEntryToOAFormDataEntry),
+    tupleToRecord
+  )
+
+const convertHoppRestReqBodyEntryToOpenApiBodyEntry = ([key, value]: [
+  string,
+  unknown
+]): [string, OpenAPIV3.SchemaObject] => {
+  return [
+    key,
+    isPlainObject(value)
+      ? {
+          properties: {
+            ...generateEntries(value as object),
+          },
+        }
+      : {
+          default: value,
+        },
+  ]
+}
+
+const generateEntries = (body: object): OpenAPIV3.SchemaObject["properties"] =>
+  pipe(
+    Object.entries(body),
+    A.map(convertHoppRestReqBodyEntryToOpenApiBodyEntry),
+    tupleToRecord
+  )
+
+const generateOpenApiHeaders = (
+  hoppHeaders: HoppRESTHeader[]
+): OpenAPIV3.ParameterObject[] =>
+  pipe(
+    hoppHeaders,
+    A.map(
+      (header): OpenAPIV3.ParameterObject => ({
+        in: "header",
+        name: header.key,
+        schema: {
+          default: header.value,
+        },
+      })
+    )
+  )
+
+const generateOpenApiQueryParams = (
+  hoppParams: HoppRESTParam[]
+): OpenAPIV3.ParameterObject[] =>
+  pipe(
+    hoppParams,
+    A.map((param) => ({
+      in: "query",
+      name: param.key,
+      schema: {
+        default: param.value,
+      },
+    }))
+  )
+
+const isValidContentType = (
+  contentType: unknown
+): contentType is keyof typeof knownContentTypes =>
+  !!(
+    contentType &&
+    typeof contentType === "string" &&
+    objHasProperty(contentType, "string")(knownContentTypes)
+  )
+
+const isNonFormDataBody = (
+  body: HoppRESTReqBody
+): body is HoppRESTReqBodyNonFormData =>
+  !!(
+    body.body &&
+    body.contentType &&
+    body.contentType !== "multipart/form-data"
+  )
+
+const isFormDataBody = (
+  body: HoppRESTReqBody
+): body is HoppRESTReqBodyFormData => body.contentType === "multipart/form-data"
+
+const isEmptyBody = (body: HoppRESTReqBody): body is HoppRestReqBodyEmpty =>
+  !(body.body && body.contentType)
+
+type RequestBodyGenerationErrors = "INVALID_CONTENT_TYPE" | "INVALID_BODY"
```

Comment:
Type names should be singular

```suggestion
type RequestBodyGenerationError = "INVALID_CONTENT_TYPE" | "INVALID_BODY"
```